### ISSUE TITLE
Alarms being cut-off and canceled 

### ISSUE THREAD

**Issue by joelm335** [NightscoutFoundation/xDrip#205]
Recently installed Xdrip+ for G5 on Samsung S6 (Android 7.0).  High/Low alarms will not sound properly.  When triggered I get a fraction of a second of sound only, so all I hear is a brief beep.  The alarms are then snoozed (without me cancelling them) for 30 mins.  Pressing the  Test Alarm button in settings also gives beep only.  I have tried choosing different alarm sound files.  

Events and Errors log shows that alarm has been triggered, then instantly dismissed and then snoozed.  I need to sort this (tried the Facebook group) because at the moment I have no effective overnight alarms.  

![logs alert screenshot](https://user-images.githubusercontent.com/32453941/31658943-6db8fd7e-b32a-11e7-84c3-8254f3c4b7a0.jpg)

**Comment by jamorham** [comment]
Can you use the `System Status` tab and let me know the xDrip version you are running. Thanks

**Comment by joelm335** [comment]
Hi:

Thanks for your really quick response:

Version is d85a06a2017.10.16

Screenshot attached.  Let me know if you need any more information/screenshots etc.  I tried posting the problem twice on the G5 Xdrip Facebook pages but nobody responded

Joel

**Comment by jamorham** [comment]
Okay that is the latest xDrip version.

I assume the problem is that something is cancelling your notifications which has android notification access. 

The most likely app to do that is Android Wear. Can you confirm whether you are using Android Wear and whether you have the latest version of that installed?  If it is installed whether you have a watch connected to it and whether you are using the xDrip watchface on the watch etc.

**Comment by joelm335** [comment]
Hi:

I have not installed Android wear and do not have an Android watch.  Not listed under settings>apps. I tried settings>apps>three dots>show system apps but it is not listed.  I only bought the phone 2 weeks ago specifically to run Xdrip and AAPS.  The only App I installed was Xdrip+.  I am normally an iPhone user and unfamiliar with Android.  On an iPhone you can search the phone – is there an equivalent on Android?

I figured from the error logs that something in the phone is cancelling notifications but couldn’t work out what.  I actually have most of the notifications blocked.  Except for Xdrip+ which is set as priority.

Joel

**Comment by joelm335** [comment]
Just searched Apps and no Android Wear or Wear App

**Comment by jamorham** [comment]
Okay, so previously when we had seen this it was Android Wear causing the problem but I don't think that would come installed by default. At the time we notified google about it but solved the problem by scheduling alerts in a different way which meant android wear didn't try to cancel them unilaterally.

On my phone in `Android Settings` -> `Sound & Notification` -> `Advanced` -> `Notification Access` it gives me a list of apps on the phone which have the permission to be able to cancel the notification which is what I think is the most likely thing to be snoozing your alert.

See if you can find something similar on your phone and look at which apps have this access level. I reckon that it will be something in there and am interested to know what the resolution is because we may need to warn other users. If its not that then there are some other things we can try but it gets more complicated.

**Comment by joelm335** [comment]
Thanks

I have gone to Settings>Notifications – this shows me that I have already turned off notification for all Apps except Xdrip.

I then went to Settings>Notifications>Advanced Notifications and it shows me the Apps that I turned off plus a whole load of additional apps, some of these are blocked and some allowed.  I have taken screenshots taken as I scrolled through the list, pasted them together and have attached the files.

I assume that it may be one of these additional Apps that are causing the problem.  We may be able to disable some of these.   I am happy to block or disable anything that is not essential for my use as a G5 receiver and pump controller. I also need basic functions for communications such as email, mms/sms etc.  Otherwise the more stuff that is not running in the background the longer my battery will last (and battery life is a major problem on this phone, which is why I already turned off most of the notifications.

Joel

**Comment by jamorham** [comment]
Ok, the attachments don't come through in your reply to the notification.  You can either email them to be directly at my username at gmail or attach them to the issue on the github website.

**Comment by joelm335** [comment]
Opened the thread on github and dragged the files into the box, but when I pressed comment I got a message “you can’t perform that action at this particular time.

I have emailed the direct to jamorham{AT}gmail.com.  Please let me know if they have arrived.

I have had another look at some of the apps in notifications>advanced.  Some of them e.g, and Gear VR service – which are shown as Notifications Allowed have actually been eating up significant amounts of my battery life (Gear VT and Knox) .  I have tried Forced Stop but still can’t get alarms to sound using the test button in Xdrip.

Joel

**Comment by joelm335** [comment]
Did you get the attachments?  Any progress?

**Comment by joelm335** [comment]
![advanced notifications 2](https://user-images.githubusercontent.com/32453941/31767573-83d6b714-b4c3-11e7-9f6d-bcc1f70714b9.jpg)
![advanced notifications 3](https://user-images.githubusercontent.com/32453941/31767575-83f090e4-b4c3-11e7-942a-8d044e246395.jpg)
![advanced notifications 1](https://user-images.githubusercontent.com/32453941/31767576-8411b260-b4c3-11e7-92ef-13f689d8509b.jpg)

**Comment by joelm335** [comment]
I have fixed the problem. The App that was snoozing the alerts was Battery Doctor (which I installed to try to improve  the battery life on the S6). Deleted the App and Xdrip alarms now working. Many thanks for your helpful suggestions.

Sent from my iPad

On 18 Oct 2017, at 11:27, Jamorham <notifications@github.com<mailto:notifications@github.com>> wrote:


Ok, the attachments don't come through in your reply to the notification. You can either email them to be directly at my username at gmail or attach them to the issue on the github website.

**Comment by jamorham** [comment]
Its a bit weird that the app would do that. Do you have a google play link for it? I could blacklist its package id to warn users.

The latest nightly also contains some attempts to work-around the issue that if an alert is cancelled within 3 seconds of being raised it tries to ignore that. 

If you wanted to assist with diagnosing this issue further you could re-install battery doctor and confirm the problem still occurs and then install the latest nightly and see whether my work-around helps at all.

Thanks.

Jon

**Comment by joelm335** [comment]
Hi John

It was definitely Battery Doctor. I uninstalled it and then opened Xdrip >settings > alarms >alerts lists > low alarm > test and it played the alarm through instead of a microsecond beep.

Will check tomorrow by reinstalling battery doctor. I am now updated to 26 Oct release. Will let you know. However dont want battery doctor anyway. Battery life better without it and no spam.

Joel
Sent from my iPhone

On 28 Oct 2017, at 21:25, Jamorham <notifications@github.com<mailto:notifications@github.com>> wrote:


Its a bit weird that the app would do that. Do you have a google play link for it? I could blacklist its package id to warn users.

The latest nightly also contains some attempts to work-around the issue that if an alert is cancelled within 3 seconds of being raised it tries to ignore that.

If you wanted to assist with diagnosing this issue further you could re-install battery doctor and confirm the problem still occurs and then install the latest nightly and see whether my work-around helps at all.

Thanks.

Jon

**Comment by joelm335** [comment]
Hi Jon:

I have now tried re-installing Battery Doctor and testing whether it is still snoozing the alarms on the 26 October release of Xdrip+.  It does not seem to be doing so.  I guess that you have solved the problem with the latest update.  FYI I have still attached a screenshot of the Google Play page.

Different question – I updated from the 16 Oct to the 26 Oct release.  With the latest release, notifications of my current sensor reading no longer appear on my locked screen if I use the OB1 collector but still do so with the G5 collector.  If I check the OB1 collector in the G5 Debug menu, the little Xdrip icon disappears from the top left of my screen.  I can use both collectors but get ~95% capture with OB1 and ~ 90% with G5 if I check scan for G5  constantly.

Thanks for all your help.

Joel
